modules/fargate/fargate.tf

```diff
@@ -0,0 +1,67 @@
+# Allow Fargate pods and EC2 workers to communicate
+
+resource "aws_security_group_rule" "eks_fargate1" {
+  count                    = var.create ? 1 : 0
+  type                     = "ingress"
+  from_port                = 0
+  to_port                  = 65535
+  protocol                 = "all"
+  security_group_id        = var.worker_security_group_id
+  source_security_group_id = var.cluster_primary_security_group_id
+}
+
+resource "aws_security_group_rule" "eks_fargate2" {
+  count                    = var.create ? 1 : 0
+  type                     = "ingress"
+  from_port                = 0
+  to_port                  = 65535
+  protocol                 = "all"
+  security_group_id        = var.cluster_primary_security_group_id
+  source_security_group_id = var.worker_security_group_id
+}
```

Comment:
See https://docs.aws.amazon.com/eks/latest/userguide/fargate-getting-started.html

> If you are working with an existing cluster that already has worker nodes associated with it, you need to make sure that pods on these nodes can communicate freely with pods running on Fargate. Pods running on Fargate are automatically configured to use the cluster security group for the cluster that they are associated with. You must ensure that any existing worker nodes in your cluster can send and receive traffic to and from the cluster security group.

Reply:
From my understanding , these are already created by the root module (https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/workers.tf#L297-L350) and https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/cluster.tf#L101-L110

Maybe I'm getting something wrong ?

Reply:
This block of code sets up the security group permissions between the default security group created by AWS and the worker groups the module creates.

This block should be moved into the main module, in a new PR, to fix https://github.com/terraform-aws-modules/terraform-aws-eks/issues/877#issuecomment-631116579 . Not sure if #877 itself was caused by this. 

Potentially breaking change as it may conflict with rules that users running a mix of "classic" workers and managed node groups or fargate could already have put in.

Reply:
I can confirm that, without these additional rules, Fargate pods are unable to communicate with pods running on "classic" worker group nodes. This does not affect managed node groups.

Reply:
arf... I see. I'm agree with @dpiddockcmp this should be moved in the root module.

@itssimon can you please do that in a separate PR ? We'll hold this until we merge that PR.

> Potentially breaking change as it may conflict with rules that users running a mix of "classic" workers and managed node groups or fargate could already have put in.

@dpiddockcmp by conflict, you mean it'll add a new rule which probably duplicates the rule added by users ?

Reply:
AWS security groups do not allow duplicated rules. The API returns an error. Only impacts users running mixed clusters who have implemented the rules exactly as above. We have no idea how many that might be.

Reply:
Done. See https://github.com/terraform-aws-modules/terraform-aws-eks/pull/892.